packages/xo-server/src/xapi/mixins/patching.js

```diff
@@ -464,4 +465,102 @@ export default {
     // host-by-host install
     throw new Error('non pool-wide install not implemented')
   },
+
+  async rollingPoolUpdate() {
+    let hosts = filter(this.objects.all, { $type: 'host' })
+    await Promise.all(hosts.map(host => host.$call('assert_can_evacuate')))
+
+    log.debug('Install patches')
+    await this.installPatches()
+
+    // Remember on which hosts the running VMs are
+    const vmsByHost = mapValues(
+      groupBy(
+        filter(this.objects.all, {
+          $type: 'VM',
+          power_state: 'Running',
+          is_control_domain: false,
+        }),
+        vm => vm.$resident_on?.$id
```

Comment:
Why? :thinking:

Reply:
Understood.

Reply:
Shouldn't we simply throw in this case?

No need to continue iterating or mapping.